addons/mail/models/mail_thread.py

```diff
@@ -2332,83 +2368,103 @@ def send_notifications():
         return True
 
     @api.model
-    def _notify_by_email_prepare_rendering_context(self, message, msg_vals, model_description=False):
+    def _notify_by_email_prepare_rendering_context(self, message, msg_vals, model_description=False,
+                                                   force_email_company=False, force_email_lang=False):
         """ Prepare rendering context for notification email.
 
         Signature: if asked a default signature is computed based on author. Either
         it has an user and we use the user's signature. Either we do not find any
         user and we compute a default one based on the author's name.
 
         Company: either there is one defined on the record (company_id field set
-        with a value), either we use env.company.
+        with a value), either we use env.company. A new parameter allows to force
+        its value.
 
         Lang: when calling this method, ``_fallback_lang`` should already been
         called, or a lang set in context with another way. A wild guess is done
         based on templates to try to retrieve the recipient's language when a flow
         like "send by email" is performed. Lang is used to try to have the
-        notification layout in the same language as the email content.
+        notification layout in the same language as the email content. A new
+        parameter allows to force its value.
 
         :param model_description: model description used in email notification process
           (computed if not given);
+        :param force_email_company: res.company record used when rendering notification
```

Comment:
Do we really need the "email" in there?
"force_company" and "force_lang" seem clear enough.

Reply:
Haha, that's where we fight. Yep I want to keep an "email" in it, because it goes through the whole call chain. You may call message_post with arguments specific to ``_notify_by_email_prepare_rendering_context`` . I want to be clear that this parameter is somehow related to email notifications. You may think that "force_lang" for example could have an effect on the body, which would be fractally wrong. Could be "force_notification_company" maybe.